add CONTRIBUTING.md (#28)
ucl-arc-tre/egress: .github/CONTRIBUTING.md

# Contributing Guidelines

> Thanks for contributing! This repository supports contributions from developers external to UCL. Because the project has access to potentially sensitive data in a Trusted Research Environment (TRE), we enforce extra security, privacy, and provenance controls. Please read this document before opening issues or pull requests.

Reviewer: ```suggestion
Thanks for contributing! This repository supports contributions from developers external to UCL. Because deployments of the project have access to potentially sensitive data in a Trusted Research Environment (TRE), we enforce extra security, privacy, and provenance controls. Please read this document before opening issues or pull requests.
```

1. Scope & expectations
- We use least-privilege, defense-in-depth, and provenance principles.
- Using sensitive data, that otherwise can only be used in a TRE, for dev/test is not permitted.

2. Legal & onboarding
<!--
- Sign the Contributor License Agreement (CLA).
-->
- Complete the short security & privacy orientation: /SECURITY_ORIENTATION.md
- By contributing you agree to the BSD 3-Clause License project license and to follow our Code of Conduct: CODE_OF_CONDUCT.md
- Enable [commit verification](https://docs.github.com/en/authentication/managing-commit-signature-verification) to ensure commit provenance.

3. Getting started
- Fork the repo and create a feature branch e.g. `feature1`.
- Run `pre-commit install` to install [pre-commit](https://pre-commit.com/).
- Run `make dev-requirements` to check all prerequisites are installed.

4. Branching & pull requests
- The `main` branch is protected. All direct pushes are disallowed.
- Open a Pull Request (PR) from your branch. PR title should be concise and reference any related issue e.g. `Fix: improve docs`.

> - Fill out the PR template with a short summary and risk assessment.

Reviewer: ```suggestion
- Fill out the PR template with required details.
```
